HearthDb/Card.cs

```diff
@@ -116,18 +129,38 @@ public string GetLocText(Locale lang)
 			var text = Entity.GetLocString(CARDTEXT_INHAND, lang)?.Replace("_", "\u00A0").Trim();
 			if(text == null)
 				return null;
-			var index = text.IndexOf('@');
-			if (index == 0)
+			var count = text.Contains("@");
+
+			if (!count && !text.Contains("|4"))
 				return text;
-			if (Entity.GetTag(PLAYER_TAG_THRESHOLD_TAG_ID) > 0)
-				return text.Substring(0, index);
-			var scriptData1 = Entity.GetTag(TAG_SCRIPT_DATA_NUM_1);
-			if (scriptData1 > 0)
-				return text.Replace("@", scriptData1.ToString());
-			var scriptData2 = Entity.GetTag(TAG_SCRIPT_DATA_NUM_2);
-			if (scriptData2 > 0)
-				return text.Replace("@", scriptData2.ToString());
-			return text.Substring(index + 1);
+
+			int? num = null;
+
+			if (SpellstoneStrings.ContainsKey(Entity.CardId))
+				return text.Replace("@", "");
+
+			if (Entity.GetTag(TAG_SCRIPT_DATA_NUM_1) != 0)
+				num = Entity.GetTag(TAG_SCRIPT_DATA_NUM_1);
+			else if (Entity.GetTag(SCORE_VALUE_1) != 0)
+				num = Entity.GetTag(SCORE_VALUE_1);
```

Comment:
I think you can simplify this all a little bit:
```
var num = Entity.GetTag(TAG_SCRIPT_DATA_NUM_1);
if (num == 0)
    num = Entity.GetTag(SCORE_VALUE_1);
if (num != 0)
    text = text.Replace("@", num.ToString());
```

0 and null are functionally equivalent here and this allows you to get rid of the nullable and separate declaration